pkg/cluster/database.go

```diff
@@ -34,7 +34,7 @@ func (c *Cluster) pgConnectionString() string {
 	password := c.systemUsers[constants.SuperuserKeyName].Password
 
 	return fmt.Sprintf("host='%s' dbname=postgres sslmode=require user='%s' password='%s' connect_timeout='%d'",
-		fmt.Sprintf("%s.%s.svc.cluster.local", c.Name, c.Namespace),
```

Comment:
While you are right, that we should not enforce `cluster.local`, we should still use a FQDN. Without the suffix we put more load on the DNS to figure out the full address itself. It's better to define the Cluster Domain for the operator and use it here.

Reply:
True. This works but will be slow at connecting because of ndots:5 and such, and I can see it causing problems down the line. It's rather annoying that Cluster Domain can't be queried from the kubernetes api.

I'll look into creating a config entry for it.

Reply:
I've [documented](https://github.com/zalando/postgres-operator/blob/master/docs/developer.md#introduce-additional-configuration-parameters) the required steps on how to introduce additional config parameters, recently. Hope it helps.